nio/client/http_client.py

```diff
@@ -374,6 +377,98 @@ def room_invite(self, room_id, user_id):
+                the server's room directory or not.
+                Defaults to ``RoomVisibility.Private``.
+
+            alias (str, optional): The desired canonical alias local part.
+                For example, if set to "foo" and the room is created on the
+                "example.com" server, the room alias will be
+                "#foo:example.com".
+
+            name (str, optional): A name to set for the room.
+
+            topic (str, optional): A topic to set for the room.
+
+            room_version (str, optional): The room version to set.
+                If not specified, the homeserver will use its default setting.
+                If a version not supported by the homeserver is specified,
+                a 400 ``M_UNSUPPORTED_ROOM_VERSION`` error will be returned.
+
+            federate (bool): Whether to allow users from other homeservers from
+                joining the room. Defaults to ``True``.
+                Cannot be changed later.
+
+            is_direct (bool): If this should be considered a
+                direct messaging room.
+                If ``True``, the server will set the ``is_direct`` flag on
+                ``m.room.member events`` sent to the users in ``invite``.
+                Defaults to ``False``.
+
+            preset (RoomPreset, optional): The selected preset will set various
+                rules for the room.
+                If unspecified, the server will choose a preset from the
+                ``visibility``: ``RoomVisibility.public`` equates to
+                ``RoomPreset.public_chat``, and
+                ``RoomVisibility.private`` equates to a
+                ``RoomPreset.private_chat``.
+
+            invite (list): A list of user id to invite to the room.
+
+            initial_state (list): A list of state event dicts to send when
+                the room is created.
+                For example, a room could be made encrypted immediatly by
+                having a ``m.room.encryption`` event dict.
+
+            power_level_override (dict): A ``m.room.power_levels content`` dict
+                to override the default.
+                The dict will be applied on top of the generated
+                ``m.room.power_levels`` event before it is sent to the room.
+        """
+
+        request = self._build_request(Api.room_create(
+            self.access_token,
+            visibility           = visibility,
+            alias                = alias,
+            name                 = name,
+            topic                = topic,
+            room_version         = room_version,
+            federate             = federate,
+            is_direct            = is_direct,
+            preset               = preset,
+            invite               = invite,
+            initial_state        = initial_state,
+            power_level_override = power_level_override,
+        ))
+
+        return self._send(request, RequestInfo(RoomCreateResponse))
+
+
```

Comment:
The convention is to put a single newline between methods inside a class. Two lines between stuff that isn't in a class. If you run make test, there is a flake8 test that is run and it complains about this.

Reply:
About flake8, could you add a config file for the project? Else, `make test` just uses whatever global config the user has on his machine. An isort file would be good too.

Reply:
Sure, lets open a separate issue for this so we can discuss what we're going to chose.

Reply:
The mentioned issue is here https://github.com/poljar/matrix-nio/issues/71.